core/include/scipp/core/comparison.h

```diff
@@ -17,8 +20,18 @@ namespace scipp::core {
 template <typename T>
 bool is_approx(const VariableConstProxy &a, const VariableConstProxy &b,
                const T tol) {
-  if (a.dtype() != b.dtype())
-    return false;
+  if (a.dtype() != b.dtype()) {
+    std::stringstream ss;
```

Comment:
Do we get any benefit from `stringstream` here? It may be cleaner to just use `operator+`.

Reply:
Good point. All arguments are strings anyway.